util/conditions/unstructured_getter.go

```diff
@@ -0,0 +1,47 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+package conditions
+
+import (
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	clusterv1 "sigs.k8s.io/cluster-api/api/v1alpha3"
+	"sigs.k8s.io/cluster-api/util"
+)
+
+// UnstructuredGetter return a Getter object that can read conditions from an Unstructured object.
+func UnstructuredGetter(u *unstructured.Unstructured) Getter {
+	return &unstructuredWrapper{Unstructured: u}
+}
+
+type unstructuredWrapper struct {
+	*unstructured.Unstructured
+}
+
+// GetConditions returns the list of conditions for an Unstructured object.
+// Please note that due to the constraints of the JSON unmarshal operations, this operation is "at best effort";
+// most specifically:
+// - it is not possible to detect if the object has an empty condition list or if it does not implement conditions;
+//   in both cases the operation returns an empty collection list is returned
+// - if the object does not implement the condition type defined in Cluster API, unmarshal matches incoming
+//   object keys to the keys; this can le to to conditions values partially set.
+// - errors during unmarshal are ignored and a empty collection list is returned.
+func (c *unstructuredWrapper) GetConditions() clusterv1.Conditions {
```

Comment:
Would be good to have a test case for this

Reply:
done